CHANGELOG.md

```diff
@@ -13,8 +13,243 @@ _Note: Gaps between patch versions are faulty, broken or test releases._
+  console.log(typeof elem.myCommand) // outputs "function"
+  ```
+  ```js
+  // v5
+  browser.addCommand('myCommand', () => { ... })
+  const elem = $('myElem')
+  console.log(typeof browser.myCommand) // outputs "function"
+  console.log(typeof elem.myCommand) // outputs "undefined"
+  elem.addCommand('myElemCommand', () => { ... })
+  const elem2 = $('myOtherElem')
+  console.log(typeof elem.myCommand) // outputs "function"
+  console.log(typeof elem2.myCommand) // outputs "undefined"
+  ```
+* custom configuration for services or reporters are now directly applied to the config list, e.g.
+  ```js
+  // ...
+  reporters: [
+    'spec',
+    [
+      'junit',
+      { outputDir: __dirname + '/junit_logs' }
+    ]
+  ],
+  // ...
+  ```
+
+#### :eyeglasses: Spec Compliancy
+* implemented parameter assertions for protocol commands
+* full W3C WebDriver compliancy
+* full Appium and Mobile JSONWire Protocol compliancy
+* simplified protocol command maintenance by defining commands, their parameters and response values within simple [json constructs](https://github.com/webdriverio/webdriverio/tree/master/packages/webdriver/protocol)
+* instead of switching protocol within a running session, WebdriverIO now determines the supported protocol by the driver based on the create session response
+
+#### :rocket: New Feature
+* new package `@wdio/applitools-service` for simple visual regression testing with [Applitools](https://applitools.com/)
+* new package `eslint-plugin-wdio` for WebdriverIO specific linting rules for [ESLint](https://eslint.org/)
+* `@wdio/devtools-service` now with frontend performance testing capabilities (see [example](https://github.com/christian-bromann/webdriverio-performance-testing))
+* new `region` [option](https://github.com/webdriverio/webdriverio/blob/cb-changelog/examples/wdio.conf.js#L29-L33) to simply run tests on SauceLabs in different datacenters
+* [`debug`](http://beta.webdriver.io/docs/api/browser/debug.html) command now allows to connect the runner with the [Chrome DevTools](https://developers.google.com/web/tools/chrome-devtools/) application for more indepth debugging
+* decoupling of `@wdio/sync` package from framework adapters so that there is no need to install [`fibers`](https://www.npmjs.com/package/fibers) when you want to run your commands with async/await
+* autofetching of all provides log types
+* auto retry mechansim for all command requests
+* auto refetch mechanism for stale elements (still [work in progress](https://github.com/webdriverio/webdriverio/pull/3118))
+* simplified reattachment to existing sessions with `attach` functionality
+* integrated and auto maintained [TypeScript](https://www.typescriptlang.org/) definitions (still [work in progress](https://github.com/webdriverio/webdriverio/pull/2862))
+* wdio testrunner fails if no spec files were found
+
+#### :bug: Bug Fix
+* fixed lost of scope when chaining elements (e.g. `$$('div')[2].$('span').getHTML()`)
```

Comment:
* fixed loss of scope when chaining elements (e.g. `$$('div')[2].$('span').getHTML()`)